packaging/requirements.py

```diff
@@ -39,7 +39,7 @@ def __init__(self, requirement_string: str) -> None:
         try:
             req = _RequirementTuple(*parse_named_requirement(requirement_string))
         except ParseExceptionError as e:
-            raise InvalidRequirement(str(e))
+            raise InvalidRequirement(str(e)) from e
```

Comment:
I'm not sure if repeating the string from `e` gets us anything. I think raising the appropriate exception is important for `except` clauses, but a person looking at the output can look at the traceback to see what ultimately triggered the exception.

```suggestion
            raise InvalidRequirement from e
```